app/viewmodels/helpers/TimelineHelper.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.helpers
+
+import models.response.emcsTfe.getMovementHistoryEvents.{GetMovementHistoryEventsResponse, MovementHistoryEvent}
+import play.api.i18n.Messages
+import utils.{DateUtils, Logging}
+import viewmodels.TimelineEvent
+
+import java.time.format.DateTimeFormatter
+import java.time.{LocalDate, LocalDateTime}
+import javax.inject.Inject
+import scala.util.Try
+
+class TimelineHelper @Inject()() extends Logging with DateUtils {
+
+  def timeline(historyEventsResponse: GetMovementHistoryEventsResponse)(implicit messages: Messages): Seq[TimelineEvent] =
+    historyEventsResponse.movementHistory.map { event =>
+      TimelineEvent(
+        eventType = event.eventType,
+        title = messages(getEventTitleKey(event)),
+        date = parseDateTime(event.eventDate),
+        url = getHistoryEventUrl(event)
+      )
+    }.sortBy(_.date)
+
+
+  def getEventTitleKey(event: MovementHistoryEvent): String = {
+    (event.eventType, event.sequenceNumber, event.messageRole) match {
+      case ("IE801", 1, 0) => s"movementHistoryEvent.${event.eventType}.first.label"
+      case ("IE801", 2, 0) => s"movementHistoryEvent.${event.eventType}.further.label"
```

Comment:
We have asked CORE for documentation on what is likely to come back for all the different events but agree it would be fine to use `_` to cater for potential higher sequence number values.

Reply:
https://github.com/hmrc/emcs-tfe-frontend/pull/50/commits/74ac828d045c5eddaa5333639045e1cc14deec22